### DISCUSSION TITLE
After Env Refresh process - Reset link

### DISCUSSION THREAD

**Discussion by ReneVolkmer** [microsoft/Dual-write-automations#11] [Q&A]
Hello community,
a customer requires a regular refresh of Support environment (FO + CE using dual write) from Production. We are able to fully automatize (via DevOps pipeline i.e.) FO and CE copy part with few DualWrite related manual steps (hitting manually "Reset link" button, stopping/starting mappings via this Dual-write automations utility UI). We are going to automatize also Stopping/starting steps, but how to deal with "Reset link"? Is there any API solution for this last missing part? Thanks for response.

Best regards
Rene

**Comment by ReneVolkmer** [comment]
@Hax1337 any hint please?

**Reply by Hax1337** [reply]
@ReneVolkmer So far the reset link is not implemented yet but can be done surely. 

Do you actually need to do the reset link or isnt applying maps and Stop / Start enough to get the env up and running? 

Also if you run already everything in pipelines, you can use this guide i posted to get the DW stuff also into the pipeline:

https://www.linkedin.com/pulse/dual-write-automations-how-to-ui-azure-pipelines-max-hentschel?trackingId=2ezRURKJjfU%2FfGyoyDy4uQ%3D%3D&lipi=urn%3Ali%3Apage%3Ad_flagship3_profile_view_base_recent_activity_content_view%3BsotYZCCkRHyFE%2BiPMjOAGA%3D%3D

**Reply by ReneVolkmer** [reply]
@Hax1337  I think Reset Link needs to be done after envs refresh, as far as I know and based on info available on the net. do you know if current API (you use by this tool) has already posibility to call "reset link" please? If not, is it in enhancements pipeline please? If not, is there any chance to put there please?

**Reply by Hax1337** [reply]
Sure it can be included in one of the next releases, I analyzed the network trace and it should be something possible. 

Should it always do a force reset or keeping the option to force or "normal" mode? @ReneVolkmer

**Reply by Hax1337** [reply]
I added it as an issue #12

**Reply by ReneVolkmer** [reply]
@Hax1337 do you have any additional info about difference between "Force" and "normal" please? There is no such a parameter when executed via standard UI (DW portal)

**Reply by ReneVolkmer** [reply]
@Hax1337  I guess you are referring to "Force reset - Forcibly stop all running maps", which is available in DW portal when any map is running (with following side note "You currently have running table maps. It is not recommended for production environments to force unlink. Table maps that are currently active may not operate properly with potential data loss in some cases. Would you like to continue with a force unlink?"). It will be good to have ""Force reset" as an option. Thank you.
